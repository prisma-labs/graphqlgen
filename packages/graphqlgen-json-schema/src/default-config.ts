import { GraphQLGenDefinition } from './definition'

export const createConfig = (
  rootDir: string = './src/graphql',
  outputFolder: string = 'generated',
) => {
  const getRootFolder = (file: string) => {
    return `${rootDir}/${file}`
  }

  const getOutputFolder = (file: string) => {
    return `${rootDir}/${outputFolder}/${file}`
  }

  const defaultConfig: GraphQLGenDefinition = {
    language: 'typescript',
    schema: getRootFolder('schema/**.graphql'),
    context: getRootFolder('context.ts:Context'),
    models: {
      files: [getRootFolder('models/**.ts')],
    },
    output: getOutputFolder('graphqlgen.ts'),
    'schema-output': getOutputFolder('graphqlgen.schema.graphql'),
    'resolver-scaffolding': {
      layout: 'file-per-type',
      output: getOutputFolder('tmp-resolvers'),
    },
  }

  return defaultConfig
}

export const defaultConfig: GraphQLGenDefinition = createConfig()
