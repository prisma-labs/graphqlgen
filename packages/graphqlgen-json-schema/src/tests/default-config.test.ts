import { createConfig, defaultConfig } from '../default-config'

test('Default config', () => {
  expect(defaultConfig).toMatchSnapshot()
})

test('Custom config', () => {
  expect(
    createConfig('./src/custom-folder', '__generated_custom__'),
  ).toMatchSnapshot()
})
