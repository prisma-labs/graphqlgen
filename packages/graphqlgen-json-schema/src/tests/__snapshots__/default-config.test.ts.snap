// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom config 1`] = `
Object {
  "context": "./src/custom-folder/context.ts:Context",
  "language": "typescript",
  "models": Object {
    "files": Array [
      "./src/custom-folder/models/**.ts",
    ],
  },
  "output": "./src/custom-folder/__generated_custom__/graphqlgen.ts",
  "resolver-scaffolding": Object {
    "layout": "file-per-type",
    "output": "./src/custom-folder/__generated_custom__/tmp-resolvers",
  },
  "schema": "./src/custom-folder/schema/**.graphql",
  "schema-output": "./src/custom-folder/__generated_custom__/graphqlgen.schema.graphql",
}
`;

exports[`Default config 1`] = `
Object {
  "context": "./src/graphql/context.ts:Context",
  "language": "typescript",
  "models": Object {
    "files": Array [
      "./src/graphql/models/**.ts",
    ],
  },
  "output": "./src/graphql/generated/graphqlgen.ts",
  "resolver-scaffolding": Object {
    "layout": "file-per-type",
    "output": "./src/graphql/generated/tmp-resolvers",
  },
  "schema": "./src/graphql/schema/**.graphql",
  "schema-output": "./src/graphql/generated/graphqlgen.schema.graphql",
}
`;
